src/ica_api.c

```diff
@@ -1071,8 +1073,10 @@ unsigned int ica_rsa_key_generate_mod_expo(ica_adapter_handle_t adapter_handle,
 	/* Keys should comply with modulus_bit_length */
 	if ((modulus_bit_length + 7) / 8 != public_key->key_length)
 		return EINVAL;
-	/* Minimum length for public exponent is sizeof(unsigned long) */
-	if (public_key->key_length < sizeof(unsigned long))
+	/* Minimum key length is sizeof(unsigned long), max key length is 4096
+	 * because of CEX adapter restriction */
+	if (public_key->key_length < sizeof(unsigned long) ||
+		public_key->key_length > MAX_RSA_KEY_BYTES)
 		return EINVAL;
```

Comment:
Would it make sense to return a specific error code for the case that the RSA key length is not supported by the HW? Similar to the specific return code when an EC curve is not supported. That way the application has a chance to detect the reason for the error.

Reply:
Good idea ... I'm now returning EPERM in case the modulus bit length is greater than 4096